### PR TITLE
include 2 methods of spacing string interpolation

### PR DESCRIPTION
**Pull request by onebree**
As per issue #370 I am including 2 padding methods for sting interpolation.

### CODE REVIEW

**Comment by bbatsov** [comment]
The second variant is used pretty rarely in practice, which is the reason it never got included in the guide.

**Comment by onebree** [comment]
If you mean the second as `"#{ user.name }`", THAT was the part in the guide. I normally see the variant I added (without space-padding)

**Comment by philoserf** [comment]
:-1:

**Comment by bbatsov** [comment]
> If you mean the second as "#{ user.name }", THAT was the part in the guide. I normally see the variant I added (without space-padding)

Ah, I totally missed this. I don't even remember when this rule got in the guide.

**Comment by onebree** [comment]
If the rule with added-padding is not used much by the community, then it should be removed from both sections of the README.

I can modify my PR to what you prefer.

**Comment by bbatsov** [comment]
Guess that'd be best, unless there are strong objections for this.

**Comment by onebree** [comment]
What if I change the wording as `good` and `okay - not preferable`? Especially since there was a downvote earlier today.

**Comment by bbatsov** [comment]
You can replace the first two sentences with something saying that interpolated expressions should have no spaces around them and add a note after the examples saying that adding a single space is an option, albeit not a popular one.

**Comment by onebree** [comment]
I just modified the strong interpolation section. Should I also edit embedded expressions where there is padded-spacing?

If there is one, I do not see a difference between string interpolation and embedded expressions.

**Comment by bbatsov** [comment]
Oh, yeah. There's none - just remove this section about embedded expressions. Guess it was placed there to be close to the rules about `{}`.

File: README.md [hunk 1]
```diff
@@ -244,7 +244,7 @@ Translations of the guide are available in the following languages:
   literals. Whichever one you pick - apply it consistently.
 
   As far as embedded expressions go, there are also two acceptable
-  options:
+  options
```
Comment (bbatsov) [inline README.md]: Why did you remove this?

Comment (onebree) [inline README.md]: I did not notice that. Must have hit backspace while reading that section. As you said in https://github.com/bbatsov/ruby-style-guide/pull/429#issuecomment-95178829 I will remove the section in general.

Comment (bbatsov) [inline README.md]: OK

File: README.md [hunk 2]
```diff
@@ -3047,13 +3031,18 @@ resource cleanup when possible.
   ```
 
 * <a name="pad-string-interpolation"></a>
```
Comment (bbatsov) [inline README.md]: This should be named simply `string-interpolation`.

File: README.md [hunk 3]
```diff
@@ -227,8 +227,8 @@ Translations of the guide are available in the following languages:
   ```
 
   `{` and `}` deserve a bit of clarification, since they are used
-  for block and hash literals, as well as embedded expressions in
-  strings. For hash literals two styles are considered acceptable.
+  for block and hash literals, as well as string interpolation.
```
Comment (bbatsov) [inline README.md]: Remove the trailing space here.

File: README.md [hunk 4]
```diff
@@ -3047,13 +3031,18 @@ resource cleanup when possible.
...
-  "#{ user.last_name }, #{ user.first_name }"
+  # good - no spaces
+  "From: #{user.first_name}, #{user.last_name}"
+ 
+  # okay - not recommended, yet arguably easier to read
+  "From: #{ user.first_name }, #{ user.last_name }"
   ```
+  
+  As in the second example, adding a _single_ space after `{` and before `}` is an option, albeit not a popular one. Like always, stick to one style throughout your code.
+<sup>[[link](#pad-string-interpolation)]</sup>
```
Comment (bbatsov) [inline README.md]: The link should be in the beginning of the rule.

**Comment by philoserf** [comment]
I am not in favor of this "Okay" version. If folks want to deviate from the community norms then go for it. The norm clearly is no spaces.

Speaking in the same language as found in the okay of the most recent read me edit, "Like, just do want you want if you aren't following the standard but don't polute the standard with uncommon cases."

I remain :-1:

**Comment by philoserf** [comment]
To clarify a bit more.  I think an "Okay" section is warranted when the community has demonstrated two competing styles regularly. If, on the other hand, the community has consistently used a style an "Okay" should not be included. Do we really want to promote a minority opinion? Such is counter, in my view, to the meaning of _standard_.

**Comment by onebree** [comment]
@philoserf Thank you for the input. I agree. If supported by @bbatsov I will remove the "okay" option. Would I then need the example at all? (Since it would be self-explanatory, to me at least).

I kept the "okay" example because of the original purpose for the PR -- there were two styles in the guide, and I wanted to address that. But I agree that pretty much everyone omits the added space padding.

Again, let me reference issue #370, which shows this problem in the guide.

**Comment by philoserf** [comment]
@onebree Thank you for the reference. I did need to re-read it. I believe the space/padding language should removed from the guide. It is both rare and only seems to serve to muddy the matter. 

What say yee, followers of issue #370 and this PR?

**Comment by bbatsov** [comment]
I'm fine with just labelling the padded version as bad and be done with it.

File: README.md [hunk 5]
```diff
@@ -3046,14 +3030,19 @@ resource cleanup when possible.
...
   ```Ruby
-  "#{ user.last_name }, #{ user.first_name }"
+  # bad - with spaces as padding
+  "From: #{ user.first_name }, #{ user.last_name }"
+  
+  # good
+  "From: #{user.first_name}, #{user.last_name}"
   ```
+  
+  As in the second example, adding a _single_ space after `{` and before `}` is an option, albeit not a popular one. Like always, stick to one style throughout your code.
```
Comment (bbatsov) [inline README.md]: This sentence has to be removed.

**Comment by bbatsov** [comment]
@onebree Remove the sentence after examples and squash the commits into one. Let's wrap this finally. Please, capitalise the final commit message.

**Comment by onebree** [comment]
Ok, will do. I will get this done by the weekend -- I need to learn rebasing. Why the strict guidelines for commits/messages, though? Just to keep the history uncluttered?

**Comment by bbatsov** [comment]
> Just to keep the history uncluttered?

Yep. 5 commits for a small change doesn't make sense to me. Most commits are WIP and have no place in the commit history.

**Comment by onebree** [comment]
Hi there. The date for the rebase was set as today. Hope that is fine.. Everything should be good now!

**Comment by bbatsov** [comment]
:+1: Thanks!

**Comment by BM5k** [comment]
:-1: for not including the WILDLY MORE READABLE `” { 1 + 1 }”` as an option. Yet another cop that will just always be disabled. :rage:

**Comment by bbatsov** [comment]
@BM5k The styles supported by cops are not bound by the guide. I'd gladly accept a PR implementing the alternative style. Whether it's WILDLY MORE READLE is pretty subjective. You seem to think so, some people will definitely disagree.

**Comment by onebree** [comment]
@BM5k I think you mean: `"#{ 1 + 1 }"`?

I would like to say that my original commits (before squashing) did include a method for padding around curly braces. After the discussion in this PR, it was decided to just include what is in the guide right now.

I think the guide is meant for the common/better practices in Ruby, where RuboCop is more flexible in letting you express your own style. As this discussion shows, using spaces is rare, meaning the guide would show the community-accepted one. But since it is still acceptable in terms of syntax, it makes sense for the RuboCop alternative.
